internal/rootcoord/root_coord.go

```diff
@@ -714,6 +720,29 @@ func (c *Core) SetDataCoord(ctx context.Context, s types.DataCoord) error {
 		return rsp.Segments, nil
 	}
 
+	c.CallUpdateSegmentStateService = func(ctx context.Context, segID typeutil.UniqueID, ss commonpb.SegmentState) (retErr error) {
+		defer func() {
+			if err := recover(); err != nil {
+				retErr = fmt.Errorf("get flushed segments from data coord panic, msg = %v", err)
+			}
+		}()
+		<-initCh
+		req := &datapb.SetSegmentStateRequest{
+			SegmentId: segID,
+			NewState:  ss,
+		}
+		resp, err := s.SetSegmentState(ctx, req)
+		if err != nil || resp.GetStatus().GetErrorCode() != commonpb.ErrorCode_Success {
+			log.Error("failed to update segment state",
+				zap.Any("request", req), zap.Any("response", resp), zap.Error(err))
+			return err
+		}
+		log.Info("successfully set segment state",
+			zap.Any("segment ID", req.GetSegmentId()),
+			zap.Any("new segment state", req.GetNewState()))
+		return nil
```

Comment:
NIT: Int64 for SegmentID, String for State.String()?

Reply:
Done.